adr/0009-bundle-media-types.md

```diff
@@ -0,0 +1,53 @@
+# 9. Bundle OCI Media Types
+
+Date: 1 July 2024 [[1]](#footnotes)
+
+## Status
+
+Accepted
+
+## Context
+
+The structure of a bundle OCI artifact is as follows:
+
+![Bundle OCI Artifact Structure](../docs/.images/uds-bundle.png)
+
+
+Note that both the bundle root manifest and the Zarf package root manifest are both OCI image manifests. The Zarf image manifest is taken directly from Zarf OCI packages, and the bundle root manifest is a new OCI image manifest that contains the Zarf image manifests as layers. This structure effectively creates a "double pointer" from the `index.json` to the layers in the bundle image manifest, and then to the Zarf image manifest.
+
+This "double pointer" is problematic because, while the [OCI spec](https://github.com/opencontainers/image-spec/blob/main/manifest.md) for image manifests doesn't necessary prohibit image manifests from pointing to image manifests, this leads to undesirable behavior in our [oras-go](https://github.com/oras-project/oras-go) client which breaks compatability across Docker's registry/v2 and GHCR OCI registries.
```

Comment:
```suggestion
This "double pointer" is problematic because, while the [OCI spec](https://github.com/opencontainers/image-spec/blob/main/manifest.md) for image manifests doesn't necessarily prohibit image manifests from pointing to image manifests, this leads to undesirable behavior in our [oras-go](https://github.com/oras-project/oras-go) client which breaks compatibility across Docker's registry/v2 and GHCR OCI registries.
```